### PR TITLE
Create trusted CA certificates file when needed

### PR DESCRIPTION
**Pull request by jhernand**
Currently the gem creates the temporary file contining the trusted CA
certificates when the 'Service' class is instantiated, and removes it
when the 'disconnect' method is called. But there are situations where
the instance is used *after* the 'disconnect' method has been called. In
those situations the trusted CA certificates won't be available, and the
connection will fail. To avoid that problem this patch changes the
'Service' class so that the temporary file is created when needed,
regardless of when the 'disconnect' method is called.

This patch addresses the following bug:

  Service order request for VM provision from template fail on SSL Certificate verification
  https://bugzilla.redhat.com/1483303

### CODE REVIEW

**Comment by jhernand** [comment]
@masayag @borod108 @pkliczewski @durandom @bdune Please review.

@bdunne We will need a new release when/if this is merged.

**Comment by oourfali** [comment]
Looks good to me

File: lib/ovirt/service.rb [hunk 1]
```diff
@@ -467,5 +464,21 @@ def full_version
...
+
+    #
+    # Returns a file object containing the trusted CA certificates. The file will be created if it
+    # doesn't exist, and it shouldn't be removed or modified by the caller.
+    #
+    # @return [File] The temporary file containing the trusted CA certificates, or nil if no custom
+    #   CA certificates are used by the connection.
+    #
+    def ca_file
+      if !@ca_file && @ca_certs
```
Comment (bdunne) [inline lib/ovirt/service.rb]: Should we return nil early if `@ca_certs` is nil?  After that, it's simple memoization.
```ruby
def ca_file
  return unless @ca_certs
  @ca_file ||= Tempfile.new('ca_file').tap do |tempfile|
    tempfile.write(@ca_certs)
    tempfile.close
  end
end
```

Comment (jhernand) [reply lib/ovirt/service.rb]: Yes, that makes sense, I am changing it, thanks @bdunne.

**Comment by jhernand** [comment]
Great, thanks @bdunne.

**Comment by jhernand** [comment]
[Bug 1483303 - Service order request for VM provision from template fail on SSL Certificate verification](https://trello.com/c/Na3nicf3/193-bug-1483303-service-order-request-for-vm-provision-from-template-fail-on-ssl-certificate-verification)

**Pull request merged**